### PR TITLE
Update sonifiables list

### PR DESCRIPTION
**Pull request by ArtInLines**
The list of sonifiables is now automatically updated and includes stocks, indices and etfs. There are a couple things to note though:

1. To reduce the amount of API requrests (and to reduce amount of unnecessary sonifiables), we currently only retrieve sonifiables of a couple handpicked exchange-markets
2. Even so, we get too many sonifiables as to show them all in the UI (I tried and the app + my IDE crashed lol). Instead, we currently only show up to 100 sonifiables. If the desired sonifiable isn't there, the user has to refine their search. (We could bring back the button to load more sonifiables and the Backend code would actually already support that, but I don't think anyone would ever use that button tbh)
3. I removed the old test-stocks, so when you test it, you will at first not see any sonifiables, but after a shot moment, the updating should be done and you should see all sonifiables again
4. If you don't want the app to update the list of sonifiables on every startup, you can go to the DataRepo and set `UPDATE_SONIFIABLES` to false.

### CODE REVIEW

File: src/main/java/util/General.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	public static String strReplace(String s, char searched, char replacement) {
+		char[] chars = s.toCharArray();
+		StringBuffer out = new StringBuffer(chars.length);
+		for (int i = 0; i < chars.length; i++) {
+			char c = chars[i];
+			out.append(c == searched ? replacement : c);
+		}
+		return out.toString();
+	}
+}
```
Comment (Rex2002) [inline src/main/java/util/General.java:13]: [it's not like String offers a replace-method or something like that xd](https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/lang/String.html#replace(char,char))

**Review by Rex2002 — APPROVED** [review]
not tested on my machine, but looks good from the browser

File: src/main/java/util/General.java (hunk 1 above)
Comment (ArtInLines) [reply src/main/java/util/General.java:13]: oh you're right, I forgot that xD

Comment (ArtInLines) [reply src/main/java/util/General.java]: Using the that interestingMethodForStringManipulation now ;) I'm sure @MalteRichert would like that too xd

Comment (MalteRichert) [reply src/main/java/util/General.java]: I love that I don't even have to say anything, my reputation is doing my work for me 🚀

**Review by MalteRichert — APPROVED** [review]
I'm the opposite of @Rex2002 : I didn't look at every file but I've tested and I'm really impressed by how smooth both the initialisation and filtering are. Great job!
One question though: do we currently cache any metadata at all or do we actually always get the latest metadata at startup? Cause that would sound like an awful lot of requests

**Comment by ArtInLines** [comment]
@MalteRichert currently we store no metadata other than name and symbol (which we can get in 1 request for all sonifiables of one exchange market). Metadata like available start- and enddates would indeed be rather costly.

**Comment by MalteRichert** [comment]
Okay, I was only wondering about name and symbol for the list. Seems sensible this way 👍

**Comment by ArtInLines** [comment]
I made the updating periodic now, meaning that we only update every x days (where x is currenty set to 5).

**Pull request merged**